geps/gep-3951/index.md

```diff
@@ -0,0 +1,360 @@
+
+## User Story
+
+[GEP-3792] defines the rationale
+for allowing out-of-cluster Gateways (OCGs)
+to participate in a
+GAMMA-compliant in-cluster service mesh,
+and the problems that must be solved
+to allow them to do so.
+This GEP defines
+an extremely minimal API
+to permit experimentation
+with OCGs and
+in-cluster mTLS meshes.
+
+Nomenclature,
+background,
+goals,
+non-goals,
+problems that must be solved,
+and some discussion
+of possible solutions to those problems
+are all included
+in [GEP-3792].
+
+[GEP-3792]: https://gateway-api.sigs.k8s.io/geps/gep-3792/
+
+## Goals of the Extra-Minimal API
+
+- Allow [Chihiro] and [Ian]
+  to configurate and operate
+  an OCG and
+  an in-cluster mTLS mesh
+  that know how to work together
+  to experiment with OCG support
+  in Gateway API.
+
+## Non-Goals
+
+- Support production use of OCGs
+  in Gateway API.
+
+- Solve all of the problems
+  defined in [GEP-3792].
+
+This is an **extra-minimal** API.
+Its purpose is to allow **experimentation**
+with OCGs and in-cluster meshes,
+**not** to provide
+a production-ready solution.
+
+Using this API in production
+is **guaranteed** to result
+in anguish, heartbreak, tears, and pain.
+
+## Overview
+
+The extra-minimal OCG API
+solves two of the [GEP-3792] problems
+in a very minimal way.
+
+- It solves the
+  [trust problem]
+  by extending
+  the Mesh and Gateway resources
+  to permit specifying
+  a _trust bundle_
+  that contains the CA certificates
+  that the OCG and the mesh
+  will use to trust each other.
+
+- It solves the
+  [discovery problem]
+  by adding a label selector
+  to the Gateway resource
+  that indicates which Routes
+  are meshed.
+
+- It does not solve the
+  [protocol problem]
+  or the
+  [outbound behavior problem].
+
+[trust problem]: https://gateway-api.sigs.k8s.io/geps/gep-3792/#1-the-trust-problem
+[protocol problem]: https://gateway-api.sigs.k8s.io/geps/gep-3792/#2-the-protocol-problem
+[discovery problem]: https://gateway-api.sigs.k8s.io/geps/gep-3792/#3-the-discovery-problem
+[outbound behavior problem]: https://gateway-api.sigs.k8s.io/geps/gep-3792/#4-the-outbound-behavior-problem
+
+### Additions to the Mesh Resource
+
+The Mesh resource
+gains an `ocg` stanza
+containing a `trustBundle` field
+that refers to a ConfigMap
+that contains the CA certificate(s)
+that the mesh should trust
+when validating connections
+from the OCG:
+
+```yaml
+...
+spec:
+  ...
+  ocg:
+    trustBundle:
+      name: ocg-trust-bundle
+      namespace: ocg-namespace
+      # Key in Configmap; defaults to "ca-bundle.crt"
+      bundleKey: ca-bundle.crt
+```
+
+#### Additions to the Gateway Resource
+
+The Gateway resource
+gains a `mesh` stanza
+containing two fields:
+
+- a `trustBundle` field
+  that refers to a ConfigMap
+  that contains the CA certificate(s)
+  that the OCG should trust
+  when validating connections
+  from meshed peers
+
+- a `labelSelector` field
+  that indicates which Routes
+  are meshed.
```

Comment:
I think we had similar comment thread on the original PR. https://github.com/kubernetes-sigs/gateway-api/pull/3894#discussion_r2196684268 

I am unclear why we need to select routes, namespace seems like it would cover 90% of the cases, and we can opt-in OR opt-out services.